rawhttp-core/src/main/java/rawhttp/core/RawHttp.java

```diff
@@ -404,4 +404,11 @@ private RequestLine verifyHost(RequestLine requestLine, RawHttpHeaders.Builder h
         }
     }
 
+    private String composeHostPort(URI uri) {
+        if ((uri.getPort() >= 0) && (uri.getPort() != 80)) {
```

Comment:
Why special-case port 80 only? If this is a `https` URI, I would expect port 43 to be used by default... should that still be added to the host header?

Reply:
The default for https is actually 443: https://tools.ietf.org/html/rfc2818#section-2.3
I suggest adding a special case for that also, after checking which scheme the uri has.